src/components/MoneyRequestReportView/MoneyRequestReportTransactionList.tsx

```diff
@@ -90,19 +89,22 @@ type SortedTransactions = {
 
 const isSortableColumnName = (key: unknown): key is SortableColumnName => !!sortableColumnNames.find((val) => val === key);
 
-const getTransactionValue = (transaction: OnyxTypes.Transaction, key: SortableColumnName) => {
-    if (key === CONST.SEARCH.TABLE_COLUMNS.DATE) {
-        const dateKey = transaction.modifiedCreated ? 'modifiedCreated' : 'created';
-        return transaction[dateKey];
+const getSortValue = (transaction: OnyxTypes.Transaction, key: SortableColumnName, reportToSort: OnyxTypes.Report) => {
+    switch (key) {
+        case CONST.SEARCH.TABLE_COLUMNS.DATE:
+            return getCreated(transaction);
+        case CONST.SEARCH.TABLE_COLUMNS.MERCHANT:
+            return getMerchant(transaction);
+        case CONST.SEARCH.TABLE_COLUMNS.CATEGORY:
+            return getCategory(transaction);
+        case CONST.SEARCH.TABLE_COLUMNS.TAG:
+            return getTag(transaction);
+        case CONST.SEARCH.TABLE_COLUMNS.TOTAL_AMOUNT:
+            return getAmount(transaction, isExpenseReport(reportToSort), transaction.reportID === CONST.REPORT.UNREPORTED_REPORT_ID);
+        default:
+            return '';
```

Comment:
Please keep sorting by the DESCRIPTION key instead of removing it, and retain the current fallback value in case a new key is supported in the future

```suggestion
        case CONST.SEARCH.TABLE_COLUMNS.DESCRIPTION:
            return Parser.htmlToText(transaction.comment?.comment ?? '');
        default:
            return transaction[key];
```